### ISSUE TITLE
DL3042 does not respect PIP_NO_CACHE_DIR

### ISSUE THREAD

**Issue by slsyy** [hadolint/hadolint#585]
- [x] This is a bug report
- [ ] This is a feature request
- [x] I searched existing issues before opening this one

### Expected behavior

No error

### Actual behavior

```
Dockerfile:54 DL3042 warning: Avoid use of cache directory with pip. Use `pip install --no-cache-dir <package>
```

### Steps to reproduce the behavior

**Dockerfile** (if relevant)

```Dockerfile
ENV PIP_NO_CACHE_DIR=1
RUN pip3 install some_package
```

The `PIP_NO_CACHE_DIR=1` is read by the `pip`, and the behavior is the same, as I would use a `--no-cache-dir` flag. I need to use that variable, because in my `Dockerfile` I use a script, which call the `pip` internally, and the env variable is the only way to disable a caching

**Comment by lorenzo** [comment]
good idea! would you like to contribute this feature?
